plugins/modules/ec2_vpc_route_table.py

```diff
@@ -524,6 +552,41 @@ def ensure_subnet_associations(connection=None, module=None, route_table=None, s
     return {'changed': changed}
 
 
+def ensure_gateway_association(connection=None, module=None, route_table=None, gateway_id=None,
+                               check_mode=None):
+    changed = False
+    filters = ansible_dict_to_boto3_filter_list({'association.gateway-id': gateway_id, 'vpc-id': route_table['VpcId']})
+    try:
+        route_tables = describe_route_tables_with_backoff(connection, Filters=filters)
+    except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+        module.fail_json_aws(e, msg="Couldn't get route tables")
+    for table in route_tables:
+        if table['RouteTableId'] is None:
+            continue
+        for a in table['Associations']:
+            if a['Main']:
+                continue
+            if a['GatewayId'] == gateway_id:
```

Comment:
Can you replace "a" in for loop with a more descriptive variable name? will definitely help someone who might work on it in future.